src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java

```diff
@@ -91,7 +95,10 @@
 			publicKey = StringUtils.isEmpty(publicKey) ? oAuth2Helper.getJwtPublicKey(kid, OAUTH_URL) : publicKey;
 			JWTVerifier verifier = oAuth2Helper.getJWTVerifier(jwt, publicKey);
 			jwt = verifier.verify(token);
-			return allowAll();
+			ArrayList<String> roles = oAuth2Helper.getRoles(jwt);
+			if(!roles.isEmpty() && !Collections.disjoint(roles, ROLES)) {
```

Comment:
I think the code needs to be reworked. The expectation is certain operations will require a specific role.

Example use cases:
* As a fhir user, I can execute certain queries
* As a fhir admin, I can execute all queries and use the web ui

For now, we only need to support that first use case. In the future when we allow the web ui to be used we might implement the 2nd.

Let's use a single value in the env var.

I would probably be better to pass the required role as an argument to this function. That way the caller can specify the role required for the operation.

Also, it's a minor thing, but there should be a space between `if` and the `(` in code.

Reply:
ok I will update the changes